nat-lab/tests/test_notification_center.py

```diff
@@ -35,23 +42,56 @@ def verify_mqtt_payload(payload):
         verify_uuid(machine_uuid)
 
 
-def run_mqtt_listener():
-    payload_queue: queue.Queue[str] = queue.Queue(1)
-
-    def on_message(_client, _userdata, message):
-        payload_queue.put(f"{message.payload.decode()}")
-
-    mqttc = mqtt.Client(client_id="reciever", protocol=mqtt.MQTTv311)
-
-    mqttc.on_message = on_message
-
-    mqttc.tls_set(certfile=None, keyfile=None, cert_reqs=ssl.CERT_NONE)
-    mqttc.connect(MQTT_BROKER_IP, port=8883, keepalive=1)
-    mqttc.subscribe("meshnet", qos=0)
-
-    mqttc.loop_start()
-
-    return payload_queue, mqttc
+async def run_mqtt_listener(
+    exit_stack,
+    connection,
+    mqtt_broker_host,
+    mqtt_broker_port,
+    mqtt_broker_user,
+    mqtt_broker_password,
+):
+    mqtt_process = await exit_stack.enter_async_context(
+        connection.create_process([
+            "python3",
+            "/opt/bin/mqtt-listener.py",
+            mqtt_broker_host,
+            mqtt_broker_port,
+            mqtt_broker_user,
+            mqtt_broker_password,
+        ]).run()
+    )
+    await sleep(1)
+    return mqtt_process
```

Comment:
Agree that previous solution was cleaner. However the data returned by `v1/notifications/token` contains domain name which should be mapped to the nat-lab mqtt broker container ip and locally it can be done via editing the host  (which might be not a good solution).